ee/hogai/graph/root/tools/search.py

```diff
@@ -126,50 +103,106 @@ class SearchTool(MaxTool):
     args_schema: type[BaseModel] = SearchToolArgs
 
-    @staticmethod
-    def _get_fts_entities(include_insight_fts: bool) -> list[str]:
-        if not include_insight_fts:
-            entities = [e for e in FTSKind if e != FTSKind.INSIGHTS]
-        else:
-            entities = list(FTSKind)
-        return [*entities, FTSKind.ALL]
-
-    async def _arun_impl(self, kind: SearchKind, query: str) -> tuple[str, dict[str, Any] | None]:
+    async def _arun_impl(self, kind: str, query: str, tool_call_id: str) -> tuple[str, ToolMessagesArtifact | None]:
         if kind == "docs":
             if not settings.INKEEP_API_KEY:
                 return "This tool is not available in this environment.", None
-            if self._has_docs_search_feature_flag():
-                return await self._search_docs(query), None
-
-        fts_entities = SearchTool._get_fts_entities(SearchTool._has_fts_search_feature_flag(self._user, self._team))
+            docs_tool = InkeepDocsSearchTool(
+                team=self._team,
+                user=self._user,
+                state=self._state,
+                config=self._config,
+                context_manager=self._context_manager,
+            )
+            return await docs_tool.execute(query, tool_call_id)
+
+        if kind == "insights" and not self._has_insights_fts_search_feature_flag():
+            insights_tool = InsightSearchTool(
+                team=self._team,
+                user=self._user,
+                state=self._state,
+                config=self._config,
+                context_manager=self._context_manager,
+            )
+            return await insights_tool.execute(query, tool_call_id)
+
+        if kind not in self._fts_entities:
+            return format_prompt_string(INVALID_ENTITY_KIND_PROMPT, kind=kind), None
+
+        entity_search_toolkit = EntitySearchTool(
+            team=self._team,
+            user=self._user,
+            state=self._state,
+            config=self._config,
+            context_manager=self._context_manager,
+        )
+        response = await entity_search_toolkit.execute(query, FTSKind(kind))
+        return response, None
 
-        if kind in fts_entities:
-            entity_search_toolkit = EntitySearchToolkit(self._team, self._user)
-            response = await entity_search_toolkit.execute(query, FTSKind(kind))
-            return response, None
-        # Used for routing
-        return "Search tool executed", SearchToolArgs(kind=kind, query=query).model_dump()
+    @property
+    def _fts_entities(self) -> list[str]:
+        entities = list(FTSKind)
+        return [*entities, FTSKind.ALL]
 
-    def _has_docs_search_feature_flag(self) -> bool:
+    def _has_insights_fts_search_feature_flag(self) -> bool:
         return posthoganalytics.feature_enabled(
-            "max-inkeep-rag-docs-search",
+            "hogai-insights-fts-search",
```

Comment:
Not sure if you did it already, just a reminder to add/edit this flag when this is merged.